docs/recipes/circleci-workflows.md

```diff
@@ -1 +1,75 @@
-# CircleCI 2.0 workflows
+# Using semantic-release with [CircleCI 2.0 workflows](https://circleci.com/docs/2.0/workflows/)
+
+![CircleCI workflows screenshot](https://github.com/circleci/circleci-docs/raw/928cbb042453ae182996ce05f4e4042b02a24634/jekyll/assets/img/docs/workflow_detail.png)
+
+Similar to Travis build stages, CircleCI 2.0 workflows are a great way to run semantic-release in an independent build step.
+We can let CircleCI take care of only running a release on master, after tests passed and running it on an image with Node 8 - your test jobs could use a completely different environment, your project may not even be written in JavaScript.
+
+Make sure to set `NPM_TOKEN` and `GH_TOKEN` in Circle's project settings UI.
+
+### Annotated example `.circleci/config.yml`
+
+```yml
+version: 2
+
+jobs:
+  # Test on Node 6
+  test_node_6:
+    docker:
+      - image: circleci/node:6
+    steps: *test_steps
+  # Test on Node 8
+  test_node_8:
+    docker:
+      - image: circleci/node:8
+    steps: *test_steps
+  # Release if both test jobs passed
+  release:
+    docker:
+      # Use Node 8 for releasing
+      - image: circleci/node:8
+    steps:
+      - checkout
+      - *restore_cache
+      - run: npm install
+      - *save_cache
+      - run: npm run semantic-release
+
+# Define when/under what conditions the jobs should be executed
+workflows:
+  version: 2
+  chromeless:
+    jobs:
+      # Two jobs to test on Node 6 and Node 8
+      - test_node_6
+      - test_node_8
+      # Release job that runs only on master if the two test jobs succeeded
+      - release:
+          requires:
+            - test_node_6
+            - test_node_8
+          filters:
+            branches:
+              only: master
```

Comment:
There is no need to filter on branch as it's done by semantic-release. It's also something that is configurable so it shouldn't be hardcoded.

Reply:
That would trigger a `release` build on every branch or PR build, spin up the container, checkout code, npm install, block the build queue, report a GitHub status check for that specific build etc, just to have semantic-release figure out that it shouldn't do anything. Why not include these 3 lines in the example that users will copy+paste?

Reply:
I agree with @pvdlg, please leave it out, it will work without. I do similar optimizations in my travis.yml files (https://github.com/octokit/rest.js/blob/34109d614958f8e423beb94c6959920e79c14862/.travis.yml#L16) but I would not make it a default trough the CLI or document it that way.

For example, if I set my default branch to something else than `master` and then copy/paste the config which only runs the release on the `master` branch, it will never work. 

It’s great to give people a starting point to make semantic-release working, but it shouldn’t be the goal to put in best practices. People who look into this are likely new to both, semantic-release & circeCI and they will run into problems anyway for which they will then create issues which increases the work load for the maintainers. And maintainers time is the scarces of resources, so when in doubt, I’d always optimize for that